wiegand/api.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/python3
+from RPi.GPIO import *
+import time
+import json
+
+class Wiegand26:
+    #Reads pin numbers for input pins from JSON
+    #Facility number is also set from JSON
+    def __init__(self):
+        config = {}
+        with open('config.json', 'r') as config_file:
+            config = json.load(config_file)
+        self.data0 = config["data0"]
+        self.data1 = config["data1"]
+        self.facility = config["facility"]
+
+    #Sets data pins to IN
+    def setup(self):
+        setwarnings(False)
+        setmode(BOARD)
+        setup(self.data0, IN)
+        setup(self.data1, IN)
+
+    # When Card returns wrong Facility 
+    class FacilityError(Exception):
+        pass
+
+    # When Card returns wrong bit parity  
+    class ReadError(Exception):
+        pass
+
+    # Loops until Card is detected and returns ID value
+    def read(self):
```

Comment:
i think this can be optimized better. will look into this myself soon and see if i can help improve some things there.

Reply:
This can be optimized once the hardware is set up. We should worry about merging this right now and experiment with optimizing later

Reply:
at the moment, this repo is about simulation. i wouldn't worry about setting up the hardware and get this merge right now. this code still seems messy to me. i wouldn't merge this if it's not cleaned up yet.